### DISCUSSION TITLE
TRV TuYa TV02

### DISCUSSION THREAD

**Discussion by osbln** [Koenkk/zigbee2mqtt#14843] [Devices]
Hi all, I'm trying to set a schedule for TuYa TV02 but I can't get an example from the instructions:

> **Schedule (composite)**
> 
> Can be set by publishing to zigbee2mqtt/FRIENDLY_NAME/set with payload {"schedule": {"week_day": VALUE, "schedule": **VALUE**}}
> 
>     week_day (enum): undefined. Allowed values: monday, tuesday, wednesday, thursday, friday, saturday, sunday
>     schedule (text): undefined.

What text gets the second value (schedule (text): undefined.)?
How could a complete payload look like?

Also I noticed that "working_day" does not work as specified "mon_sun, mon_fri+sat+sun, separate" but only 0,1,2,3 is accepted.
Which value corresponds to "mon_fri+sat+sun"?

Data of my TRV:

Friendly name
    thermostat1
Description

Last seen
    N/A
Availability
    [Disabled](https://www.zigbee2mqtt.io/guide/configuration/device-availability.html#availability-advanced-configuration)
Device type
    EndDevice
Zigbee Model
    TS0601
Zigbee Manufacturer
    _TZE200_hue3yfsn
Description
    Thermostat radiator valve
Support status

    Supported
IEEE Address
    0x84fd27fffea24cc3
Network address
    0x7E2C
Manufacturer
    [TuYa](https://www.zigbee2mqtt.io/supported-devices/#v=TuYa)
Model
    [TV02-Zigbee](https://www.zigbee2mqtt.io/devices/TV02-Zigbee.html#tuya-tv02-zigbee)

thank you for your help

**Comment by dafinkom** [comment]
I am also struggling with this problem. I think the last time it worked was in version 1.27

**Comment by osbln** [comment]
i have updated from 1.28.0 to 1.28.2 and it seems to work now.

[https://github.com/Koenkk/zigbee-herdsman-converters/pull/4750](https://github.com/Koenkk/zigbee-herdsman-converters/pull/4750)

> schedule
> Schedule will work with "auto" preset. In this mode, the device executes a preset week programming temperature time and temperature. Before using these properties, check `working_day` property. Each day can contain up to 10 segments. At least 1 segment should be defined. Different count of segments can be defined for each day, e.g., 3 segments for Monday, 5 segments for Thursday, etc. It should be defined in the following format: `hours:minutes/temperature`. Minutes can be only tens, i.e., 00, 10, 20, 30, 40, 50. Segments should be divided by space symbol. Each day should end with the last segment of 24:00. Examples: `04:00/20 08:30/22 10:10/18 18:40/24 22:50/19.5`; `06:00/21.5 17:20/26 24:00/18`. The temperature will be set from the beginning/start of one period and until the next period, e.g., `04:00/20 24:00/22` means that from 00:00 to 04:00 temperature will be 20 degrees and from 04:00 to 00:00 temperature will be 22 degrees.

it seems to work with this new example

`{"schedule": {"week_day":"monday", "schedule":"06:00/22 08:00/18 18:00/20 19:30/22 23:00/19.0"}}`

**Comment by haegar33** [comment]
A possible stupid follow-on question for this device: How do I set the date and time for the thermostat? There are no MQTT commands for this. Is date/time automatically set by the Zigbee hub and Z2M?

**Comment by Giexperience** [comment]
Hi. Hello. We Raul. I am the manufacturer of TV02. thank you for your support. Our new TV06 is on the market this year. Can you write 'device support' for me? I will send the product to you. Thank you so much.
giexsmart@126.com
![主图2](https://github.com/Koenkk/zigbee2mqtt/assets/121998503/e9d064b2-a417-47d1-97be-6ac1f5d40277)
![1正方形](https://github.com/Koenkk/zigbee2mqtt/assets/121998503/5441a1df-ba6c-45ad-b02c-e642916a398e)

**Reply by Giexperience** [reply]
Hello everyone. This product is now fully compatible with Z2M. I have already submitted the update to ZHA, and it will be supported after the ZHA update in September.
